modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/parse/MBObjectStops.java

```diff
@@ -95,6 +95,46 @@
         return zoomLevels;
     }
 
+    public static MBLayer reducePropertyAndZoomFunctions(MBLayer layer) {
+        JSONObject jsonObject = (JSONObject) layer.getJson();
+        MBLayer newLayer = MBLayer.create(jsonObject);
+
+        if (newLayer.getPaint() != null) {
+            getReducedProperty(newLayer.getPaint());
+        }
+
+        return newLayer;
+    }
+
+    static JSONObject getReducedProperty(JSONObject jsonObject) {
+        Set<?> keySet = jsonObject.keySet();
+        Iterator<?> keys = keySet.iterator();
+        JSONArray reducedStops = null;
+        while (keys.hasNext()) {
+            String key = (String)keys.next();
+            if (jsonObject.get(key) instanceof JSONObject) {
+                JSONObject child = (JSONObject) jsonObject.get(key);
+                if (child.containsKey("stops")) {
+                    JSONArray stops = (JSONArray) child.get("stops");
+                    for (int i = 0; i < stops.size(); i++) {
+                        JSONArray stop = (JSONArray) stops.get(i);
+                        if (stop.get(0) instanceof JSONObject) {
+                            reducedStops = new JSONArray();
+                            JSONArray x = (JSONArray) stops.get(i);
+                            reducedStops.add(0, ((Long)((JSONObject)x.get(0)).get("zoom")).longValue());
```

Comment:
I'm not sure I understand. Is this supposed to be reducing a zoom-and-property function to a property function? (But it's using the "zoom" here, not the property "value").